javaslang/src/main/java/javaslang/collection/Array.java

```diff
@@ -1048,11 +1048,15 @@ public int length() {
 
     @Override
     public Array<T> reverse() {
-        final Object[] arr = new Object[delegate.length];
-        for (int i = 0; i < delegate.length; i++) {
-            arr[delegate.length - 1 - i] = delegate[i];
+        if (size() <= 1) {
+            return this;
+        } else {
+            final Object[] arr = new Object[delegate.length];
+            for (int i = 0; i < delegate.length; i++) {
+                arr[delegate.length - 1 - i] = delegate[i];
+            }
```

Comment:
@paplorinc 

> and using i and j in the declaration order:

highly subjective but I don't mind :) could be also seen as 1st read from source 2nd write to target. It is just subjective

The length in `i < delegate.length` is read from heap. When introducing `final int length = delegate.length`, length is put on the stack. The loop accesses length n times, so it might matter.